src/os/ObjectStore.h

```diff
@@ -452,7 +452,26 @@ class ObjectStore {
     list<Context *> on_commit;
     list<Context *> on_applied_sync;
 
+    bufferptr trans_buff;
   public:
+     
+     void set_trans_buff(bufferptr bp) {
+       trans_buff = bp;
+     }
+
+     bufferptr get_trans_buff() {
+       return trans_buff;
+     }
+
+     void update_trans_buff(unsigned len) {
+       if (trans_buff.get_raw()) {
+         if (trans_buff.offset() + len > trans_buff.raw_length())
+           trans_buff.set_length(trans_buff.raw_length());
+         else
+           trans_buff.set_length(trans_buff.length() + len);
+         assert(trans_buff.offset() == 0);
+       }
+     }
```

Comment:
I think a simpler way to do this would be to make the bufferptr refer to the *unused* portion of the buffer that is available.  Then, keep a *pointer* to the external append buffer in the bufferlist.  When we use it, we update the bufferptr (to the remaining space).  That way you don't need the fragile update_trans_buff() indicating how much was used.